providers/oidc.go

```diff
@@ -196,12 +199,37 @@ func getOIDCHeader(accessToken string) http.Header {
 
 func findClaimsFromIDToken(idToken *oidc.IDToken, accessToken string, profileURL string) (*OIDCClaims, error) {
 
-	// Extract custom claims.
 	claims := &OIDCClaims{}
-	if err := idToken.Claims(claims); err != nil {
+
+	if err := idToken.Claims(&claims.rawClaims); err != nil {
 		return nil, fmt.Errorf("failed to parse id_token claims: %v", err)
 	}
 
+	// TODO: Implement Aggregated and/or Distributed claims fetching...
+	// <https://openid.net/specs/openid-connect-core-1_0.html#AggregatedDistributedClaims>
+	// This is necessary if, for example, the identity is a part of so many groups that it can't fit
+	// them into a normal claims assertion. Unfortunately, it's entirely possible that the OAuth2
+	// provider has a lot of these that we don't really care about, so when implementing this, we
+	// probably want a config option like `OAUTH2_PROXY_FETCH_DISTRIBUTED_CLAIMS=groups` (or
+	// whatever) so that if it comes back distributed, we have the extra hint that it's actually
+	// worth the time spent fetching it before running validation. If the validator doesn't care,
+	// then we don't really care either.
+
+	// Extract custom claims manually (to avoid parsing twice)
+
+	if value, ok := claims.rawClaims["sub"].(string); ok {
+		claims.Subject = value
+	}
+	if value, ok := claims.rawClaims["email"].(string); ok {
+		claims.Email = value
+	}
+	if value, ok := claims.rawClaims["email_verified"].(*bool); ok {
+		claims.EmailVerified = value
+	}
+	if value, ok := claims.rawClaims["preferred_username"].(string); ok {
+		claims.PreferredUsername = value
+	}
```

Comment:
I can see this being prone to causing bugs in the future as this doesn't get updated. I'd prefer if there were some other way this could be done, I don't mind calling `idToken.Claims` twice personally, I don't think it's that expensive

Reply:
These particular entries won't be changing in OIDC's case, as per the spec: https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims